ui/src/components/Document/DocumentTitle.js

```diff
@@ -1,19 +1,54 @@
 import { DocumentEdition } from '@components/Document';
 import React from 'react';
+import PropTypes from 'prop-types';
 import Truncate from 'react-truncate';
+import _get from 'lodash/get';
 
-export const DocumentTitle = ({ document, short, truncate }) => {
-  const documentMetadata = document.metadata ? document.metadata : document;
-  const title = short ? (
-    documentMetadata.title
-  ) : (
-    <>
-      {documentMetadata.title} (<DocumentEdition document={document} />{' '}
-      {documentMetadata.publication_year})
-    </>
-  );
+const EditionTitleCmp = ({ metadata }) => {
+  const edition = _get(metadata, 'edition');
+  const year = _get(metadata, 'publication_year');
+  /* render both edition and year, or only edition, or only year or nothing
+   * title (edition - year)
+   * title (edition)
+   * title (year)
+   * title
+   */
+  const editionYearCmp =
+    edition && year ? (
+      <>
+        {'('}
+        <DocumentEdition metadata={metadata} /> - {year}
+        {')'}
+      </>
+    ) : edition ? (
+      <>
+        {'('}
+        <DocumentEdition metadata={metadata} />
+        {')'}
+      </>
+    ) : year ? (
+      `(${year})`
+    ) : (
+      ''
+    );
+  return editionYearCmp;
+};
+
+export const DocumentTitle = ({ metadata, titleOnly, truncate }) => {
+  const title = _get(metadata, 'title', 'No title set!');
```

Comment:
This should never happen since it is mandatory in the schema and you cannot add/update records without it, neither during migration. If it happens, you should really check!